cg/services/run_devices/pacbio/run_validator/pacbio_run_validator.py

```
@@ -57,6 +58,12 @@ def ensure_post_processing_can_start(self, run_data: PacBioRunData) -> None:
         self._touch_is_validated(run_data.full_path)
         LOG.debug(f"Run for {run_data.full_path} is validated.")
 
+    def validate_run_files(self, run_data: PacBioRunData):
```

This basically just checks whether all files are present without trying to parse them or anything. First files it checks for is the "transferdone" or manifest files.

Just be aware, is a transfer is never fully completed it will always fail silently with each time-based iteration and adding this behaviour makes it possible that "corrupted" runs will not notify production. The old approach by trying and failing each time would make it noticable. Downside of course is notification fatigue.